### DISCUSSION TITLE
Cookie Consent Language

### DISCUSSION THREAD

**Discussion by vfacta** [quarto-dev/quarto-cli#5613] [Q&A] [answered]
Is it possible to change the language of the cookie consent banner to the website's main language? 

[#433](https://github.com/quarto-dev/quarto-cli/issues/433#issue-1172994895) already asked if it was possible to change the the text inside the pop-up, but that question seems to aim at modifying the matter of what's being said there.

I don't intend to change the content per se of what is being said, but  to display the  translation from the default English text to the language set in the yml file. As the Cookie pop-up has, when clicked on "Change my preferences", a box for language selection close to the upper right corner, it was expected to have an automatic translation according to what was defined in the yml, as it is to other predefined texts.

I don't know if this resource has already been implemented or if I'm doing something wrong. 

Minimal working example:

 - _quarto.yml
```
project:
  type: website

lang: pt

website:
  
  title: Untitled

  cookie-consent: true

format:
  html: default
```

- index.qmd

```
---
title: "Lorem Cookie"
---

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```

**Comment by cderv** [comment]
I don't think we currently allow to change the language and the built in feature uses the default language which is English. 

@dragonstyle do you think it is just a matter of configuration ? It seems the js file we are bundling has different language, so it could be only a matter of supporting the configuration 
https://github.com/quarto-dev/quarto-cli/blob/cae5cb4f274c9e3b2961e270558c887b3953c17e/src/project/types/website/website-analytics.ts#L312-L335

@vfacta cookie consent tooling are using JS script and you can also probably use your own by adding some includes to HTML header to use the library of your choice for cookie consent. Just sharing as a workaround.

**Reply by vfacta** [reply]
Yes, I've seen some alternatives using JS, but it would be a nice addition if this was straight out of the box. 

Also, Quarto is great, thanks for the work of you all!

**Reply by dragonstyle** [reply]
It is just a matter of configuration and I will add support for it to Quarto 1.4!

**Reply by dragonstyle** [reply, marked as answer]
It'll be available in our next pre-release build later this AM! (https://github.com/quarto-dev/quarto-cli/commit/179078bfd8d6d41ea64b0f3a66da88856d9d4e64)
